File: Tests/GamifiedTests/ExperienceTests.swift
//
//  ExperienceTests.swift
//  
//
//  Created by Nathan Fallet on 25/03/2023.
//

import XCTest
@testable import Gamified

final class ExperienceTests: XCTestCase {
    
    func testExperienceLevel1() throws {
        let experience = Experience(total: 7)
        XCTAssertEqual(experience.level, 1)
        XCTAssertEqual(experience.current, 0)
        XCTAssertEqual(experience.toNextLevel, 9)
    }
    
    func testExperienceLevel10() throws {
        let experience = Experience(total: 161)
        XCTAssertEqual(experience.level, 10)
        XCTAssertEqual(experience.current, 1)
        XCTAssertEqual(experience.toNextLevel, 27)
    }
    
    func testExperienceLevel40() throws {
        let experience = Experience(total: 2930)
        XCTAssertEqual(experience.level, 40)
        XCTAssertEqual(experience.current, 10)
        XCTAssertEqual(experience.toNextLevel, 202)
    }
    
}

File: Gamified/src/test/java/me/nathanfallet/gamified/ExperienceTests.kt
package me.nathanfallet.gamified

import me.nathanfallet.gamified.models.Experience
import org.junit.Assert
import org.junit.Test

class ExperienceTests {

    @Test
    fun testExperienceLevel1() {
        val experience = Experience(7)
        Assert.assertEquals(1, experience.level)
        Assert.assertEquals(0, experience.current)
        Assert.assertEquals(9, experience.toNextLevel)
    }

    @Test
    fun testExperienceLevel10() {
        val experience = Experience(161)
        Assert.assertEquals(10, experience.level)
        Assert.assertEquals(1, experience.current)
        Assert.assertEquals(27, experience.toNextLevel)
    }

    @Test
    fun testExperienceLevel40() {
        val experience = Experience(2930)
        Assert.assertEquals(40, experience.level)
        Assert.assertEquals(10, experience.current)
        Assert.assertEquals(202, experience.toNextLevel)
    }

}